ocdata/adsorptions.py

```diff
@@ -601,8 +610,10 @@ def add_adsorbate_onto_surface(surface, adsorbate, bond_indices):
     # Then pick one from the reasonable configurations list as an output.
     reasonable_adsorbed_surfaces = [surface for surface in adsorbed_surfaces
                                     if is_config_reasonable(surface) is True]
-    adsorbed_surface = random.choice(reasonable_adsorbed_surfaces)
-    return adsorbed_surface
+    reasonable_adsorbed_surface_index = np.random.choice(len(reasonable_adsorbed_surfaces))
```

Comment:
I was wondering if we can modify this a little bit (not sure how much software effort it would be based on the refactored code): right now this is designed to be randomly select a site on a surface, but would be great if we can add some sort of agrs (e.g.) `sample=True` args in add_adsorbate_onto_surface. When it's sample=False, it generates all the configuration of the adsorbate on a surface. This would come in handy for the ML validation, as well as the large scale prediction we are going to make on 11K materials. We can chat more on this.

Reply:
That makes sense, it should be fairly easy to generate all the configurations. 

I'm wondering if we should change the directory structure of the outputs - right now, if we have adsorbate_index=1, bulk_index=2, surface_index=3, the files ares are saved in `1_2_surface3/surface/` and `1_2_surface3/adsorbed_surface/`. If we enumerate all the configurations, I could keep the same structure and just append to the dir name to get `[adsorbate_index]_[bulk_index]_surface[surface_index]_config[config_index]/`, but that will be a lot of files from all the combinations. Alternatively, I could make a nested directory of `[adsorbate_index]/[bulk_index]/[surface_index]/[config_index]/`. I'm not sure how the files will be read later - would it be better to have a flat or nested directory structure?

Reply:
For me, the latter makes more sense. For each adsorbate on a given surface, we will have multiple adslabs (adsorbate adsorbed on a slab),  but we will only need to relax that slab by itself once. So it seems like having all of them in one parent folder `1_2_surface3/` would be nice. I.e., `1_2_surface3/surface/` and `1_2_surface3/adsorbed_surface/config_index/`.  Maybe @sidgoyal78 can chime in as well?

Reply:
Good point, that sounds reasonable to me.

Reply:
Yeah, i think that folder structure sounds good to me.

However, in order to run jobs on the internal infra, we will need the input folders in a flat structure. Something like 
```
<prefix_dir_structure>/ 
       structure1/ 
       structure2/
        . 

       . 
       structure40000/
```
       
Just an FYI.

Reply:
This flat structure could be enforced on our side (FAIR's side) when we prepare the inputs. What do you think @janiceblue ?